### DISCUSSION TITLE
Why only train set is used for cross validation and test set is left out in pycaret?

### DISCUSSION THREAD

**Discussion by dryogibansal** [pycaret/pycaret#1689] [Q&A] [answered]
I have a doubt why 10-fold cross validation is done only on train set because we generally need test set also for cross validation?

**Comment by dryogibansal** [comment]
For training ML models, we either split data into train/test or do cross validation on the complete dataset as splitting method is considered inefficient. So, why in pycaret, all of the evaluation metrics are cross-validated results based on training set (70%) only. Could anyone please clear why test set is not included in cross validation process during training models and evaluating metrics?

**Comment by moezali1** [comment, marked as answer]
When you execute the `setup` function in PyCaret it splits the data into train and test sets (70/30) by default. Cross-validation is then done on train set only.

The hold-out set is there just for an additional sense of surety. Sometimes you may cause leakage in cross-validation and hence hold-out may reflect that issue if it occurs. However, you shouldn't use a hold-out set for any kind of performance feedback loop during the model selection phase.

Hope this clarifies your question.

**Reply by dryogibansal** [reply]
Thanks.

**Reply by makisgrammenos** [reply]
Yeah but doesnt's that give false metrics that don't correspond to reality?
If we run `finalize_model` it will train the model on the whole dataset and the real metrics will be different than the training ones.
This can be checked out if you run cross validate the model manually using  sklearn  `cross_val_score`

**Comment by H13U1999** [comment]
I also have a question, when we do Cross-Validation in pycaret, is the validation-set the same size as test set ?
